src/utils/registerNewNetwork.ts

```diff
@@ -73,39 +75,144 @@ export const registerNewNetwork = async (
     },
     blockTime: arbitrumSdkConstants.ARB_MINIMUM_BLOCK_TIME_IN_SECONDS,
   };
+}
 
-  let isL1NetworkRegistered = false;
-  try {
-    await getL1Network(l1Provider);
-    isL1NetworkRegistered = true;
-  } catch {
-    isL1NetworkRegistered = false;
+async function createParentNetwork<TIsArbitrum extends boolean>({
+  parentNetwork,
+  childNetwork,
+  isArbitrum,
+}: {
+  parentNetwork: Network;
+  childNetwork: Network;
+  isArbitrum: TIsArbitrum;
+}): Promise<TIsArbitrum extends true ? L2Network : L1Network>;
+async function createParentNetwork<TIsArbitrum>({
+  parentNetwork,
+  childNetwork,
+  isArbitrum,
+}: {
+  parentNetwork: Network;
+  childNetwork: Network;
+  isArbitrum: TIsArbitrum;
+}): Promise<L1Network | L2Network> {
+  if (isArbitrum) {
+    return {
+      chainID: parentNetwork.chainId,
+      name: parentNetwork.name,
+      confirmPeriodBlocks: 0,
```

Comment:
my bad here, even though this should never be used let's still put in a reasonable value
```suggestion
      confirmPeriodBlocks: 45818,
```